src/coreclr/vm/jitinterface.cpp

```diff
@@ -11822,6 +11873,74 @@ void* CEEJitInfo::getFieldAddress(CORINFO_FIELD_HANDLE fieldHnd,
     return result;
 }
 
+bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t* buffer, int bufferSize)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    _ASSERT(fieldHnd != NULL);
+    _ASSERT(buffer != NULL);
+    _ASSERT(bufferSize > 0);
+
+    bool result = false;
+
+    JIT_TO_EE_TRANSITION();
+
+    FieldDesc* field = (FieldDesc*)fieldHnd;
+    if (field->IsStatic() && !field->IsThreadStatic() && ((unsigned)bufferSize == field->GetSize()))
+    {
+        MethodTable* pEnclosingMT = field->GetEnclosingMethodTable();
+        if (!pEnclosingMT->IsSharedByGenericInstantiations())
+        {
+            // Allocate space for the local class if necessary, but don't trigger
+            // class construction.
+            DomainLocalModule* pLocalModule = pEnclosingMT->GetDomainLocalModule();
+            pLocalModule->PopulateClass(pEnclosingMT);
+
+            GCX_COOP();
+            if (pEnclosingMT->IsClassInited() && IsFdInitOnly(field->GetAttributes()))
+            {
+                if (field->IsObjRef())
+                {
+                    OBJECTREF fieldObj = field->GetStaticOBJECTREF();
+                    if (fieldObj != NULL)
+                    {
+                        Object* obj = OBJECTREFToObject(fieldObj);
+                        if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(obj))
+                        {
+                            intptr_t ptr = (intptr_t)obj;
+                            memcpy(buffer, &ptr, sizeof(intptr_t));
+                            result = true;
+                        }
+                    }
+                    else
+                    {
+                        memset(buffer, 0, sizeof(intptr_t));
+                        result = true;
+                    }
+                }
+                else
+                {
+                    void* fldAddr = field->GetStaticAddressHandle(field->IsRVA() ? nullptr : (void*)field->GetBase());
+                    if (fldAddr != nullptr)
+                    {
```

Comment:
```suggestion
                    _ASSERTE(fldAddr != nullptr);
```
This should never return null